src/missed_spans.rs

```diff
@@ -278,11 +278,10 @@ impl<'a> FmtVisitor<'a> {
                     let first_line = &subslice[..offset];
                     self.push_str(first_line);
                     self.push_str(&comment_indent.to_string_with_newline(self.config));
-
-                    let other_lines = &subslice[offset + 1..];
+                    let other_lines = &subslice[offset + 1..].trim_start();
                     let comment_str =
                         rewrite_comment(other_lines, false, comment_shape, self.config)
-                            .unwrap_or_else(|| String::from(other_lines));
+                            .unwrap_or_else(|| other_lines.to_string());
```

Comment:
Does it make sense to trim the start of the comment if we fail to rewrite it? By the time we call `unwrap_or_else` `other_lines` has already been trimmed. Maybe we should leave the string alone if `rewrite_comment` fails?

```rust
let other_lines = &subslice[offset + 1..];
let comment_str = 
    rewrite_comment(other_lines.trim_start(), false, comment_shape, self.config)
        .unwrap_or_else(|| other_lines.to_string());
```

Reply:
Agree this is better. Changed.